volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -33,8 +37,10 @@ def get_requirements(cls):
             ),
         ]
 
-    def _generator(self):
-        layer = self.context.layers[self.config["primary"]]
+    def enumerate_mft_records(self, attr_callback):
+        phys_layer = self.context.layers[self.config["primary"]].config["memory_layer"]
```

Comment:
@ikelos I can catch keyerror around this but I am confused:

 1) how it would fail?
 2) what would the backup be to get access to the physical layer? Returning None would mean we can't recover records from the physical layer, which would be a bug we need to fix.

Is there example code of how to get the physical layer if the key error hits?

Reply:
Well, it's mostly if `primary` is found but a) isn't a layer or b) isn't an intel layer (for some reason, possibly in the future) and therefore doesn't contain a `memory_layer`.  That could happen if the config is loaded from a file and could send us chasing down the wrong path.  This code chains several assumed config items together without any error handling which is a bit brittle.  It'll throw an exception but that's not really a good result for the user.  Chances are we'll never run into them, but handling errors will help us figure out things going wrong that we didn't expect down the line.  The backup would be a nice error message informing the user that something went wrong (or better, how it went wrong).

It's not a huge deal and there's lots of other places where we don't do proper error checking for all conditions, but at a minimum I'd prefer it to be split into at least two lines, so that we can figure out which lookup is failing...

Reply:
I split them up and added wording to request the user to create an issue on the tracker if those lines are triggered since it seems like it would be some major internal bug, so hopefully we can catch and fix it.